lib/AST/ASTImporter.cpp

```diff
@@ -4505,10 +4524,17 @@ Decl *ASTNodeImporter::VisitClassTemplateDecl(ClassTemplateDecl *D) {
 
   ToTemplated->setDescribedClassTemplate(D2);
 
+  assert(
+    (!ToTemplated->getPreviousDecl() ||
+    ToTemplated->getPreviousDecl()->getDescribedClassTemplate() == PrevDecl)
+    && "Experimental check failed");
+
+  if (PrevDecl)
+    D2->setPreviousDecl(PrevDecl);
   D2->setAccess(D->getAccess());
   D2->setLexicalDeclContext(LexicalDC);
-  ToTemplated->setLexicalDeclContext(LexicalDC);
-  LexicalDC->addDeclInternal(D2);
+  if (!IsFriend)
```

Comment:
This is something where I am not sure.

Reply:
That looks okay to me. 

Because:
We cannot define a friend class in the befriending class. Consequently there must be at lease one declaration in the enclosing scope of the befriending class. During the import of that declaration we will add that to the DC.